File: HillHigh1980.Core/ApplicationService/IRosterService.cs
using HillHigh1980.Core.Entity;
using HillHigh1980.Core.Entity.Jut.Locations;
using HillHigh1980.Core.Entity.Jut.Rosters;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace HillHigh1980.Core.ApplicationService
{
    public interface IRosterService
    {
        Task<RosterJut> FindRosterByIdAsync(int seedId);
        Task<List<RosterJut>> FindRostersByName(Filter filter);
        Task<List<RosterJut>> GetAllRostersAsync();
        Task<int> CreateRosterLocation(List<LocationJut> locations);
        Task<List<LocationJut>> GetRosterLocations(int rosterId);
        Task<LocationJut> UpdateRosterLocation(LocationJut location);
        Task RemoveRosterLocation(LocationJut location);


    }
}

File: HillHigh1980.UI/wwwroot/Script/Core/ApplicationService/IRosterService.ts
module GScope {
    export module ApplicationService {
        export interface IRosterService {
            FindRosterById(rosterId: number): Promise<Entity.Roster>;
            FindRostersByName(filter: Entity.Filter): Promise<Entity.Roster[]>;
            GetAllRosters(): Promise<Entity.Roster[]>;
            CreateRosterLocations(locations: Entity.Location): Promise<any>;
            UpdateRosterLocation(location: Entity.Location): Promise<any>;
            DeleteRosterLocation(location: Entity.Location): Promise<any>;
        }
    }
}